xarray/core/nputils.py

```diff
@@ -28,11 +28,11 @@
 try:
     import numbagg
 
-    _HAS_NUMBAGG = Version(numbagg.__version__) >= Version("0.5.0")
+    _NUMBAGG_VERSION: Version | None = Version(numbagg.__version__)
```

Comment:
In the DuckArrayModule which checks for existance of modules, this is solved like this:

https://github.com/pydata/xarray/blob/04eb3420ff2664339c6b9b28d03ca622fae05fcd/xarray/core/pycompat.py#L55

Reply:
Yeah, I'm open to that. It does have the disadvantage that we have two variables rather than one, and one of them is "wrong" (i.e. we don't have `0.0.0` installed). But would you prefer that way?

Reply:
I don't follow. I think that if you modify DuckArrayModule slightly to handle numbagg: 
https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/core/pycompat.py#L19

And used `mod_version("numbagg")` instead I think it would be easier to read:
https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/core/pycompat.py#L81

This will enable importing numbagg lazily.

I think numbagg should not be imported eagerly when xarray is imported, could you add numbagg to this list?
https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/tests/test_plugins.py#L216-L240

Reply:
Just use this one? https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/namedarray/utils.py#L42

Reply:
We can set `.type` to `()` — lmk if that's not OK...

https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/core/pycompat.py#L76

Reply:
> the import should be closer to that of Zarr  

I agree. Let's just use https://github.com/pydata/xarray/blob/cfe4d71fae70930ac6776bd53fe2a93875a84515/xarray/namedarray/utils.py#L42 and move on. It's infinitely less confusing